src/bookmarks/bookmarkfiltermodel.cpp

```diff
@@ -0,0 +1,30 @@
+#include "bookmarkfiltermodel.h"
+
+BookmarkFilterModel::BookmarkFilterModel(QObject *parent)
+    : QSortFilterProxyModel(parent)
+{
+
+}
+
+int BookmarkFilterModel::getIndex(int currentIndex)
+{
+    QModelIndex proxyIndex = index(currentIndex, 0);
+    QModelIndex sourceIndex = mapToSource(proxyIndex);
+    return sourceIndex.row();
+}
+
+bool BookmarkFilterModel::filterAcceptsRow(int sourceRow, const QModelIndex &sourceParent) const
+{
+    bool contains = false;
+    QModelIndex index = sourceModel()->index(sourceRow, 0, sourceParent);
+
+    for (int i = 0; i < sourceModel()->roleNames().count(); i++) {
+        int it = i + static_cast<int>(Qt::UserRole);
+        QString roleName = sourceModel()->roleNames().value(it);
+        if (roleName == "url" || roleName == "title") {
```

Comment:
I would prefer this way:
```suggestion
        if (roleName == QLatin1String("url") || roleName == QLatin1String("title")) {
```
Also I think it's possible to search once for these two role indexes by their names and then just check data only for these two found role values.

Reply:
Since role ids are fixed at compile time, they could be simply queried directly from the source model. If this filter model is supposed to work with any source model that has `"url"` and `"title"` roles (which I doubt), then indices should be queried when the model is being set. But that's when performance matters. In case of browser it obviously doesn't :)